### PR TITLE
Collector fix and buyback script

### PR DESCRIPTION
**Pull request by anhdungle93**
fix the collector contract and add tests. All four tests are very similar just for different types of pairs: token-token, token-wADA, PT-wADA, token-PT.

### CODE REVIEW

File: scripts/buyback.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+async function main() {
+
+    let collectorAddress;
+
+    if (hre.network.name == "milkomedaTestnet") {
+        collectorAddress = "0x92A76FE5e70F4C9d44F6BD126ce61BFFB6563320";
+    } else if (hre.network.name == "milkomedaMainnet") {
+        collectorAddress = "0x2324797D029E7192e62a4e758e8Ca3Aae74BF1EB";
+    }
```
Comment (zwilling) [inline scripts/buyback.js:13]: For me it was helpful in the past to have an else case throwing an error if another network is used.
It just makes the error simpler to understand than when the code below fails with an unknown collectorAddress.

File: scripts/setLockInCollector.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    /* await CollectorInstance.setLock(false); */
+    console.log(await CollectorInstance.locked());
+    await sleep(15);
+    
+    /* await sleep(120);
+    await hre.run("verify:verify", {
+        address: TokenInstance.address,
+        constructorArguments: [],
+    }); */
```
Comment (zwilling) [inline scripts/setLockInCollector.js:37]: This commented code can be removed.

File: test/collector.test.ts [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+        1000000000000000
+      );
+
+    const collectorLTBalanceAfter = await tokenPTPair.balanceOf(
+      collector.address
+    );
+    // fee LT should be transferred to the collector contract
+    expect(await collectorLTBalanceAfter).to.be.gt(collectorLTBalanceBefore);
+
+    // now we buy back protocol token using the LT of coin1 and coin2 inside the collector
```
Comment (zwilling) [inline test/collector.test.ts:466]: The buy back part in the tests is quite repetitive. I suggest to move it to a function.

**Review by zwilling — APPROVED** [review]
@anhdungle93 Looks good. I only have minor code style comments for this PR. We can merge it already if it is urgent. If you want to fix these small comments, ping me afterwards and I merge it.

It is important that we also apply this fix to the repository forked from OccamX. 
@mike-tis Please assign this issue to someone: https://redboot.atlassian.net/browse/BLOCKCHAIN-456

I noticed two small bugs in the tests that are unrelated to the collector. I'll create a small fix PR for it later.

File: contracts/Collector.sol [hunk 4]
```diff
@@ -236,6 +236,7 @@ contract Collector is Ownable {
         returns (uint256 amountOut)
     {
         // X1 - X5: OK
```
[the remark this answers is not in the document]
Comment (anhdungle93) [reply contracts/Collector.sol:238]: I don't know and have removed them. They are from sushiswap repo where I copied the contract from.

File: scripts/buyback.js (hunk 1 above)
Comment (anhdungle93) [reply scripts/buyback.js]: I have add the error of unsupported networks to the buyback and also other scripts if necessary

File: scripts/setLockInCollector.js (hunk 2 above)
Comment (anhdungle93) [reply scripts/setLockInCollector.js]: removed

File: test/collector.test.ts (hunk 3 above)
Comment (anhdungle93) [reply test/collector.test.ts:466]: If there is a need to write more test or change this part in the future I will move it to a separate function.

**Review by zwilling — APPROVED** [review]
Nice

**Pull request merged**